### PR TITLE
[mp] 🦄 🪄 mage magic mission: delightful documentation for `dataengineering.wiki`

### PR DESCRIPTION
**Pull request by mattppal**
This provides some background and basic info on Mage. I checked the table rendering with Obsidian and it looks good. 

I started from the formatting for existing tools and added a bit of flourish. I'm happy to adjust this if there's a style we need to conform to.

**Disclaimer**: I'm employed by Mage, but I did my best to maintain an objective take in the Pros/Cons. If there are any other items to be added (or removed), I'm happy to make adjustments.

### CODE REVIEW

File: Tools/Workflow Orchestrators/Mage.md [hunk 1]
```diff
@@ -1,14 +1,84 @@
...
+- **Interactive code**: Immediately see results from your code’s output with an interactive notebook UI.
+- **Data is a first-class citizen**: Each block of code in your pipeline produces data that can be versioned, partitioned, and cataloged for future use.
+- **Collaborate on cloud**: Develop collaboratively on cloud resources, version control with Git, and test pipelines without waiting for an available shared staging environment.
+- **Scaling made simple**: Transform very large datasets directly in your data warehouse or through a native integration with Spark.
+- **Observability**: Operationalize your pipelines with built-in monitoring, alerting, and observability through an intuitive UI.
+- **Rapidly growing community**: Mage has a vibrant community of over 2.5k data professionals as of 07/23.
+- **Data integration**: Use existing connectors or build your own with the Singer-spec for a free alternative to paid tools, like Fivetran. Full table and incremental via CDC (change data capture) support.
+- **Native integration with dbt**: preview dbt results, orchestrate dbt model runs, schedule dbt models to depend on non-dbt tasks (e.g. ETL/ELT pipelines).
```
Comment (JPHaus) [inline Tools/Workflow Orchestrators/Mage.md:66]:
```suggestion
- **Hybrid GUI/Code Tool:** design-driven GUI for building and editing pipelines while still allowing the flexibility of code.
- **Engineering best practices built-in:** each step in your pipeline is a standalone file containing modular code that’s reusable and testable with data validations.
- **Interactive code**: Immediately see results from your code’s output with an interactive notebook UI.
- **Data is a first-class citizen**: Each block of code in your pipeline produces data that can be versioned, partitioned, and cataloged for future use.
- **Collaborate on cloud**: Develop collaboratively on cloud resources, version control with Git, and test pipelines without waiting for an available shared staging environment.
- **Scaling made simple**: Transform very large datasets directly in your data warehouse or through a native integration with Spark.
- **Data integration**: Use existing connectors or build your own with the Singer-spec for a free alternative to paid tools, like Fivetran. Full table and incremental via CDC (change data capture) support.
- **Native integration with dbt**: preview dbt results, orchestrate dbt model runs, schedule dbt models to depend on non-dbt tasks (e.g. ETL/ELT pipelines).
```
I will preface these suggestions with the fact that I am not super familiar with Mage. I just felt a few bullets were subjective while others are kind of expected. For example, if all of the workflow orchestrators have observability then it's not really an advantage that needs to be listed. I could be wrong though!

File: Tools/Workflow Orchestrators/Mage.md [hunk 2]
```diff
@@ -1,14 +1,84 @@
...
+
+## Disadvantages
+- **Requires frequent patches**: Frequent releases/version upgrades mean that some maintenance is required.
+- **Work in progress**: version 0.9.3 as of 07/23— this is not yet a v1 tool.
+- **No managed offering**: there is only a self-hosted option, currently.
+- **Compliance**: At this time, Mage only supports SOC-2 security standards.
```
Comment (JPHaus) [inline Tools/Workflow Orchestrators/Mage.md:72]: Is there security higher than SOC-2? I thought that was pretty standard.

**Review by JPHaus — APPROVED** [review]
@mattppal First of all congrats on making your first contribution to the wiki! Open to hearing feedback on how we can improve the experience for contributors. I approved the PR and will merge it but wanted to give some time for you to check out some of my comments first. Overall, this is fantastic and I really appreciate your attention to detail. Thank you!

File: Tools/Workflow Orchestrators/Mage.md (hunk 2 above)
Comment (mattppal) [reply Tools/Workflow Orchestrators/Mage.md:72]: I guess this isn't very specific. Mage _does not_ have the ability to support HIPAA or GDPR regulations just yet, which is what I was getting at.

File: Tools/Workflow Orchestrators/Mage.md [hunk 3]
```diff
@@ -1,14 +1,84 @@
...
+- **Data is a first-class citizen**: Each block of code in your pipeline produces data that can be versioned, partitioned, and cataloged for future use.
+- **Collaborate on cloud**: Develop collaboratively on cloud resources, version control with Git, and test pipelines without waiting for an available shared staging environment.
+- **Scaling made simple**: Transform very large datasets directly in your data warehouse or through a native integration with Spark.
+- **Observability**: Operationalize your pipelines with built-in monitoring, alerting, and observability through an intuitive UI.
+- **Rapidly growing community**: Mage has a vibrant community of over 2.5k data professionals as of 07/23.
+- **Data integration**: Use existing connectors or build your own with the Singer-spec for a free alternative to paid tools, like Fivetran. Full table and incremental via CDC (change data capture) support.
+- **Native integration with dbt**: preview dbt results, orchestrate dbt model runs, schedule dbt models to depend on non-dbt tasks (e.g. ETL/ELT pipelines).
+
+## Disadvantages
+- **Requires frequent patches**: Frequent releases/version upgrades mean that some maintenance is required.
```
[the remark this answers is not in the document]
Comment (mattppal) [reply Tools/Workflow Orchestrators/Mage.md:69]: image

This was inspired by an actual comment in our slack 😂

File: Tools/Workflow Orchestrators/Mage.md (hunk 1 above)
Comment (mattppal) [reply Tools/Workflow Orchestrators/Mage.md:66]: I suppose we think about observability a bit differently, but I'm fine with dropping this one

**Pull request merged**